### PR TITLE
Improve loading sources

### PR DESCRIPTION
**Pull request by jasonLaster**
Fixes Issue: #4849

### Summary of Changes

Previously, loading perf.html when it used webpack `eval-source-map` took about 17 seconds of heavy work. Now it takes about 4 seconds of less intensive work and feels pretty snappy.

* adds some checks from #5645, which prevent unnecessary updates
* batches all of the calls to `newSources` from `loadSourceMap`

Loading is now pretty snappy

NOTE: this also works for cheap-source-map

#### Before

- lots of loadSourceMap calls
- lots of updateTree calls

#### After

- a few updateTree calls
- minimal loadSourceMap time

### CODE REVIEW

File: src/actions/sources/newSources.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ import { loadSourceText } from "./loadSourceText";
 import { togglePrettyPrint } from "./prettyPrint";
 import { selectLocation } from "../sources";
 import { getRawSourceURL, isPrettyURL } from "../../utils/source";
-
+import { flatten } from "lodash";
```
Comment (AlexisDeschamps) [inline src/actions/sources/newSources.js]: This seems very cool.

File: src/actions/sources/newSources.js [hunk 2]
```diff
@@ -171,11 +179,10 @@ export function newSources(sources: Source[]) {
       dispatch(checkPendingBreakpoints(source.id));
     }
 
-    await Promise.all(
-      filteredSources.map(source => dispatch(loadSourceMap(source.id)))
-    );
+    await dispatch(loadSourceMaps(filteredSources));
```
Comment (AnshulMalik) [inline src/actions/sources/newSources.js]: We should load source maps for only the generated sources, how about we incorporate that change here as well?

Comment (darkwing) [reply src/actions/sources/newSources.js]: Thoughts on this @jasonLaster or can we merge?

Comment (jasonLaster) [reply src/actions/sources/newSources.js]: done. good idea!

**Pull request merged**